Feat: Add API Delete call to the button

edumargra opened the pull request:
Straightforward, added API calls to the button and controlled the response based on the satatus code.

No tests added because _time._
Since the motos API isn't updated on the frontend, I couldn't check with the frontend the error 401 of the API, but since it's tested on the backend and passes I believe is okay.

edumargra committed:
Add API DELETE call

rchacoma8 commented:
> If the user doesn't exist, show message: It seems that an account with the email ${this.email} doesn't exist.
Este punto lo puedo testear desde frontend 

>If the user has a motorbike reserved/running, show message: You cannnot delete an account with a motorbike reserved/running!
En esta rama, hay error en el metodo GET de motos. Es necesario hacer merge de develop (ya que ahora está solucionado en develop)

> Add a general message error: Internal Server error, try again or get in touch with customer support
No puedo comprobar desde frontend cuándo se generaría.

rchacoma8 commented:
Si la password es incorrecta, aparece el mensaje indicado - [x]
Cuando se introduce la password correcta, elimina la cuenta (status code 200) y redirecciona a la home page - [x]
Al intentar entrar de nuevo, salta el dialog indicando que el usuario o la contraseña son incorrectos, ok - [x]

rchacoma8 commented:
> 
    > If the user has a motorbike reserved/running, show message: You cannnot delete an account with a motorbike reserved/running!
En esta rama, hay error en el metodo GET de motos. Es necesario hacer merge de develop (ya que ahora está solucionado en develop)

Después de la observación de mi compañero, lo pruebo en local.

Sí puedo eliminar una cuenta teniendo una moto reservada

![Captura de pantalla de 2020-12-15 18-16-47](https://user-images.githubusercontent.com/32159059/102248904-0b7a9700-3f02-11eb-8706-fbb04eafe870.png)
![Captura de pantalla de 2020-12-15 18-18-53](https://user-images.githubusercontent.com/32159059/102248907-0cabc400-3f02-11eb-8540-0b5e9806043e.png)

rchacoma8 commented:
No sé si tengo el código del backend en esta rama o en develop.

Pero según veo, frontend debería poder pulsar el botón, hacer la llamada a la api y ser el backend el que comprueba si tiene motos reservadas o activas. 

Lo que quiero decir, es que el comportamiento de frontend es el esperado, ya que atrapa el error en un switch-case.

Ahora investigaré o veré a quién le corresponde revisar el backend.

rchacoma8 commented:
OK, el código de backend está, pero **no funciona como esperado**.
@edumargra habla con quien corresponda.

edumargra commented:
entendido @rchacoma8 , gracias! Ahora se lo comento a @SrLimoncio

rchacoma8 commented:
Como el código de frontend tiene el comportamiento esperado (suponiendo un comportamiento correcto), apruebo el pull request.

edumargra committed:
Fix: change error message

API had different behavior than I thought, if there is a motorbike reserved, I can delete the account.

edumargra commented:
@rchacoma8 Lo he hablado con @SrLimoncio el creador de la API y @edumargra PO

Habia entendido mal lo que se esperaba de l'aplicacion cuando el cliente tiene una moto reservada. En este caso, *SI* que puede eliminarla.

He cambiado la _issue associada_ y el mensaje de 401.

Disculpa las molestias.
